Lib/test/test_typing.py

```diff
@@ -6004,6 +6004,8 @@ def test_mutablesequence(self):
         self.assertNotIsInstance((), typing.MutableSequence)
 
     def test_bytestring(self):
+        # ensure it's not already imported
+        typing._remove_cached_ByteString_from_globals()
```

Comment:
Why is this necessary? Isn't it called at the end of every test run by `clear_caches` here?

https://github.com/python/cpython/blob/a052be4c0d9707ebdb935b8314ed22e8d3d99fc0/Lib/test/libregrtest/utils.py#L198-L204

Reply:
It possibly isn't quite necessary, but it made my life easier when I thought the cleanup itself was causing the refleak. Also, another test running earlier in the same process could have triggered the import. I'll take it out of this PR though as it's not directly related.

Reply:
The `clear_caches()` function should run at the end of _every_ test, not just `test_typing`. My understanding was that it was meant to make it so that we didn't have to worry about this kind of stuff in most test files :)

Reply:
OK. This test still feels kind of fragile though: it will fail if we add a test earlier in typing.py that accesses `ByteString`.